applications/serial_lte_modem/src/http_c/slm_at_httpc.c

```diff
@@ -262,15 +263,19 @@ static int do_http_connect(void)
 			ret = -errno;
 			goto exit_cli;
 		}
-		ret = setsockopt(httpc.fd, SOL_TLS, TLS_PEER_VERIFY, &peer_verify,
-				 sizeof(peer_verify));
+		ret = setsockopt(httpc.fd, SOL_TLS, TLS_PEER_VERIFY, &httpc.peer_verify,
+				 sizeof(httpc.peer_verify));
 		if (ret) {
 			LOG_ERR("setsockopt(TLS_PEER_VERIFY) error: %d", -errno);
 			ret = -errno;
 			goto exit_cli;
 		}
-		ret = setsockopt(httpc.fd, SOL_TLS, TLS_HOSTNAME, httpc.host,
-				 strlen(httpc.host));
+		if (httpc.hostname_verify) {
+			ret = setsockopt(httpc.fd, SOL_TLS, TLS_HOSTNAME, httpc.host,
+					 strlen(httpc.host));
+		} else {
+			ret = setsockopt(httpc.fd, SOL_TLS, TLS_HOSTNAME, NULL, 0);
```

Comment:
Is this really required?
Would it be same as just not set the hostname at all?

Reply:
Seems you're right.
> NRF_SO_SEC_HOSTNAME
Set the hostname used for peer verification. The option value is a null-terminated string containing the host name to verify against. The option length is the size in bytes of the hostname. Passing NULL as an option value and 0 as an option length disables peer hostname verification. By default, peer hostname verification is disabled.

Reply:
This is unfortunately required as our test servers do not have valid certificate names (CN / Subject Alt Name). When TLS stack is in modem side, modem will accept the certificates even without the socket option set to NULL. With native TLS, the CN / Subject Alt Name is checked to exist in the certificate, if the option is not set to NULL. Setting the option to NULL completely skips the hostname validation.

Reply:
Ah, so the default behavior is not the same with native TLS.